pkg/openstack/glance.go

```diff
@@ -71,32 +75,36 @@ func ReconcileGlance(ctx context.Context, instance *corev1beta1.OpenStackControl
 		instance.Spec.Glance.Template.GlanceAPIs[name] = glanceAPI
 	}
 
-	if glance.Status.Conditions.IsTrue(glancev1.GlanceAPIReadyCondition) {
-		// initialize the main APIOverride struct
-		if instance.Spec.Glance.APIOverride == nil {
-			instance.Spec.Glance.APIOverride = map[string]corev1beta1.Override{}
-		}
+	// initialize the main APIOverride struct
+	if instance.Spec.Glance.APIOverride == nil {
+		instance.Spec.Glance.APIOverride = map[string]corev1beta1.Override{}
+	}
 
-		var changed bool = false
-		for name, glanceAPI := range instance.Spec.Glance.Template.GlanceAPIs {
-			if _, ok := instance.Spec.Glance.APIOverride[name]; !ok {
-				instance.Spec.Glance.APIOverride[name] = corev1beta1.Override{}
-			}
-			// Retrieve the services by Label and filter on glanceAPI: for
-			// each instance we should get **only** the associated `SVCs`
-			// and not the whole list. As per the Glance design doc we know
-			// that a given instance name is made in the form: "<service>
-			// <apiName> <apiType>", so we build the filter accordingly
-			// to resolve the label as <service>-<apiName>
-			svcs, err := service.GetServicesListWithLabel(
-				ctx,
-				helper,
-				instance.Namespace,
+	var changed = false
+	for name, glanceAPI := range instance.Spec.Glance.Template.GlanceAPIs {
+		if _, ok := instance.Spec.Glance.APIOverride[name]; !ok {
+			instance.Spec.Glance.APIOverride[name] = corev1beta1.Override{}
+		}
+		// Retrieve the services by Label and filter on glanceAPI: for
+		// each instance we should get **only** the associated `SVCs`
+		// and not the whole list. As per the Glance design doc we know
+		// that a given instance name is made in the form: "<service>
+		// <apiName> <apiType>", so we build the filter accordingly
+		// to resolve the label as <service>-<apiName>
+		svcs, err := service.GetServicesListWithLabel(
+			ctx,
+			helper,
+			instance.Namespace,
+			util.MergeMaps(
+				GetServiceOpenStackOperatorLabel(glance.Name),
```

Comment:
This should allow to filter by `APIName` but also using the `ooSelector` that has been added to extend the label selector, right?

Reply:
yes, as mentioned above this is to ignore other "internal" services created by the service operators.